src/cns.ts

```diff
@@ -190,6 +192,35 @@ export default class Cns extends EthereumNamingService {
     return await this.record(domain, 'ipfs.redirect_domain.value');
   }
 
+  /**
+   * Gets all keys that are setuped in the domain.
+   * If domain has attached one of the legacyResolvers it will look only for a standarDKeyLists
+   * to see if they are set. It won't be able to parse the custom keys.
+   * @param domain - domain name
+   */
+  async getAllKeys(domain: string): Promise<any> {
+    const tokenId = this.namehash(domain);
+    const resolver = await this.getResolver(tokenId);
+    if (!resolver) throw new ResolutionError(ResolutionErrorCode.UnspecifiedResolver, {domain})
+    const resolverContract = this.buildContract(resolverInterface, resolver);
+    if (isLegacyResolver(resolver)) {
+      return await this.getStandardKeys(resolverContract, tokenId);
+    }
+    const logs = await resolverContract.fetchLogs('NewKey', tokenId);
+    const keyHashes = logs.map(event => event.topics[2]);
+    const result = await this.callMethod(resolverContract, 'getManyByHash', [keyHashes, tokenId]);
+
+    return result;
+  }
+
+  private async getStandardKeys(resolverContract: Contract, tokenId: string) {
+    const keys = Object.values(standardKeys);
+    const response = await this.callMethod(resolverContract, 'getMany', [keys, tokenId]);
+    return keys.map((value, index) => {
+      if (response[index]) return value;
+    }).filter(val => val);
```

Comment:
You can do this with just `filter`, there's no need to map to undefined values first and then filtering them out